data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/codec/NewlineDelimitedCodec.java

```diff
@@ -42,18 +47,34 @@ public void parse(final InputStream inputStream, final Consumer<Record<Event>> e
     }
 
     private void parseBufferedReader(final BufferedReader reader, final Consumer<Record<Event>> eventConsumer) throws IOException {
+        final boolean doAddHeaderToOutgoingEvents = Objects.nonNull(this.headerDestination);
```

Comment:
What should be the expected behavior if  headerDestination is empty String `""`?
nit: you can remove this keyword in line 50 & 73 to access `headerDestination`.